internal/controller/dnsrecord_controller.go

```diff
@@ -221,21 +229,14 @@ func setDNSRecordCondition(dnsRecord *v1alpha1.DNSRecord, conditionType string,
 func (r *DNSRecordReconciler) applyChanges(ctx context.Context, dnsRecord *v1alpha1.DNSRecord, managedZone *v1alpha1.ManagedZone, isDelete bool) error {
 	logger := log.FromContext(ctx)
 
-	rootDomain, err := dnsRecord.GetRootDomain()
-	if err != nil {
-		return err
-	}
-	if !strings.HasSuffix(rootDomain, managedZone.Spec.DomainName) {
-		return fmt.Errorf("inconsitent domains, does not match managedzone, got %s, expected suffix %s", rootDomain, managedZone.Spec.DomainName)
-	}
-	rootDomainFilter := externaldnsendpoint.NewDomainFilter([]string{rootDomain})
+	rootDomainFilter := externaldnsendpoint.NewDomainFilter([]string{managedZone.Spec.DomainName})
```

Comment:
We shouldn't set this to `managedZone.Spec.DomainName` all the time. 

If we have a rootHost set in the spec, we should be setting it to that and falling back to `managedZone.Spec.DomainName` when it's not set. The filters here are how we are going to ensure that records returned when querying the provider are only records related to this recordset (Endpoints array) See note here https://github.com/Kuadrant/dns-operator/blob/main/internal/controller/dnsrecord_controller.go#L290 This will be really important when we start work on the distributed DNS plan as we can't have other records coming back from the provider.

re. Wildcards, rootDomain here should probably exclude the `*.` prefix if it has one.